test_full_sync failing

test_full_sync is failing:
```
 replica_info = await replica_client.info("ALL")
>           pytest.fail(
                f"Replica did not sync in time. \nmaster: {master_info} \n\nreplica: {replica_info}"
            )
E           Failed: Replica did not sync in time. 
```

https://github.com/dragonflydb/dragonfly/actions/runs/16233894101/job/45841203328

--- a/src/server/tiering/disk_storage.cc
+++ b/src/server/tiering/disk_storage.cc
@@ -180,9 +180,9 @@ std::error_code DiskStorage::Stash(io::Bytes bytes, StashCb cb) {
   if ((available < 256_MB) && (available < capacity * 0.15) && !grow_pending_) {
     auto ec = TryGrow(256_MB);
     LOG_IF(ERROR, ec && ec != errc::file_too_large) << "Could not call grow :" << ec.message();
-    return ec;
   }
-  return {};
+
+  return {};  // Must succeed after the operation was scheduled to run cleanup
 }
 
 DiskStorage::Stats DiskStorage::GetStats() const {

